planning/costmap_generator/CMakeLists.txt

```diff
@@ -47,6 +47,13 @@ rclcpp_components_register_node(costmap_generator_node
 
 if(BUILD_TESTING)
   find_package(ament_cmake_ros REQUIRED)
+  ament_add_ros_isolated_gtest(test_objects_to_costmap
+    test/test_objects_to_costmap.cpp
+  )
+
+  target_link_libraries(test_objects_to_costmap
```

Comment:
current code links costmap_generator_lib already

Reply:
Using the humble docker image `ghcr.io/autowarefoundation/autoware-universe:humble-latest-cuda` I could reproduce the linker error and here is what I could understand.
- Function `fromMsg` is declared but undefined (`U` symbol when using `nm`) in the `costmap_generator_lib`.
- This does not cause issue for the node as it is only built as a library and as a plugin, meaning this undefined function is expected to come from another library at runtime.
- For the test, we build an executable which requires all functions to be defined -> error.
```console
$ nm -gDC build/costmap_generator/libcostmap_generator_lib.so | grep fromMsg
  U tf2::fromMsg(geometry_msgs::msg::Quaternion_<std::allocator<void> > const&, tf2::Quaternion&)`
```

### Solution

Including `<tf2_geometry_msgs/tf2_geometry_msgs.hpp>` in `objects_to_costmap.hpp` solves the issue and `fromMsg` no longer appears as symbol in `libcostmap_generator_lib.so`. I think this is because the `fromMsg` function declared in `tf2_geometry_msgs.hpp` is inline and is optimized away during compilation.

I pushed a commit with the change 88a941cbe

Reply:
how do you create your container?

Reply:
I followed the instructions from [the documentation](https://autowarefoundation.github.io/autoware-documentation/main/installation/autoware/docker-installation/).
```sh
rocker --nvidia --x11 --user --volume $HOME/autoware -- ghcr.io/autowarefoundation/autoware-universe:humble-latest-cuda
```
I had issues with missing dependencies inside the docker image which I resolved with `rosdep`.
```sh
rosdep init && sudo apt update && rosdep install -iry --from-paths src --rosdistro $ROS_DISTRO
```
I also had to checkout the branch `ros2` in  `src/universe/external/grid_map` before I could successfully build up to the `costmap_generator`.
```sh
colcon build --packages-up-to costmap_generator -DCMAKE_BUILD_TYPE=Release --symlink-install
```

Reply:
the only difference is that i create docker container by 
```
docker run --name aw_humble -v /home/sc/work/autoware.humble:/autoware -it  ghcr.io/autowarefoundation/autoware-universe:humble-latest-cuda
```

BUT,follow your steps,i create a new contaniner,tested codes without include tf header, still can not reproduce link error....

my steps:
```
rocker --nvidia --x11 --user --volume /home/sc/work/autoware.humble/:/autoware -- ghcr.io/autowarefoundation/autoware-universe:humble-latest-cuda
sudo apt-get update
sudo apt-get install ros-humble-nav2-msgs
checkout grid_map to branch ros2
colcon build --packages-up-to costmap_generator --cmake-args -DCMAKE_BUILD_TYPE=Release --symlink-install
```

by the way
```
colcon build --packages-up-to costmap_generator -DCMAKE_BUILD_TYPE=Release --symlink-install
```
should be 
```
colcon build --packages-up-to costmap_generator --cmake-args -DCMAKE_BUILD_TYPE=Release --symlink-install
```

Since humble ci can pass after add include tf header file, I won't spend  more time on debuging why no link error on my local container.